third-party/realdds/src/dds-stream.cpp

```diff
@@ -90,15 +95,30 @@ void dds_stream::start_streaming()
 
 void dds_video_stream::handle_data()
 {
-    topics::image_msg frame;
     dds_sample sample;
-    while( _reader && topics::image_msg::take_next( *_reader, &frame, &sample ) )
+    if( _compressed )
     {
-        if( ! frame.is_valid() )
-            continue;
-
-        if( is_streaming() && _on_data_available )
-            _on_data_available( std::move( frame ), std::move( sample ) );
+        topics::compressed_image_msg frame;
+        while( _reader && topics::compressed_image_msg::take_next( *_reader, &frame, &sample ) )
```

Comment:
Can't we merge this 2 flows into a single one somehow?

Reply:
As discussed, I will move the logic to a template function in the h file, `handle_data` will call it with appropriate type.